pkg/clusteragent/admission/mutate/agent_sidecar/util.go

```diff
@@ -116,3 +117,79 @@ func applyFargateOverrides(container *corev1.Container) {
 		Value: "true",
 	})
 }
+
+////////////////////////////////
+//                            //
+//     Profiles Overrides     //
+//                            //
+////////////////////////////////
+
+// ProfileOverride represents environment variables and resource requirements overrides
+type ProfileOverride struct {
+	EnvVars              []corev1.EnvVar             `json:"env,omitempty"`
+	ResourceRequirements corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+func applyProfileOverrides(container *corev1.Container) {
+	if container == nil {
+		return
+	}
+
+	// Read and parse profiles
+	profilesJSON := config.Datadog.GetString("admission_controller.agent_sidecar.profiles")
+
+	var profiles []ProfileOverride
+
+	err := json.Unmarshal([]byte(profilesJSON), &profiles)
+	if err != nil {
+		log.Errorf("failed to parse profiles for admission controller agent sidecar injection: %s", err)
+		return
+	}
+
+	if len(profiles) > 1 {
```

Comment:
Your understanding is correct.

Since `profiles` and `selectors` are different concepts, I am basing the decision of registering (or refusing to register) the webhook on the provided selectors. A valid selector config implies a successful registration. Likewise, `profiles` define how to override. Thus I am basing the decision to override (or rejecting overrides) on the provided profiles.

I am treating these 2 decisions separately.

I am ok with no registering the webhook unless we have at most 1 selector and at most 1 profile.

WDYT?

In addition, if decide not to register the webhook if the user provides more than 1 profile, what if the user provides profiles which are malformed (not parseable to json). Should we register the webhook in that case?

Reply:
If user provides profile override which can't be parse, is incorrect for other reasons, or not allowed (like multiple overrides) and we still register webhook what could happen is that we inject default container (we start with default and then apply overrides) - which clearly isn't what user wanted (because they provided override) and they may not discover it right away because thing somehow works.
So, I think input validation should be a prerequisite of webhook registration.

Reply:
agreed 👍